lidor/src/mapping.ts

```diff
@@ -0,0 +1,122 @@
+  }
+
+  // BigInt and BigDecimal math are supported
+  entity.count = entity.count + BigInt.fromI32(1)
+
+  // Entity fields can be set based on event parameters
+  entity.account = event.params.account
+
+  // Entities can be written to the store with `.save()`
+  entity.save()
+
+  // Note: If a handler doesn't require existing field values, it is faster
+  // _not_ to load the entity from the store. Instead, create it fresh with
+  // `new Entity(...)`, set the fields that should be updated and save the
+  // entity back to the store. Fields that were not set or unset remain
+  // unchanged, allowing for partial updates to be applied.
+
+  // It is also possible to access smart contracts from mappings. For
+  // example, the contract that has emitted the event can be connected to
+  // with:
+  //
+  // let contract = Contract.bind(event.address)
+  //
+  // The following functions can then be called on this contract to access
+  // state variables and other data:
+  //
+  // - contract.addIdentityAdmin(...)
+  // - contract.addrToDID(...)
+  // - contract.authenticationPeriod(...)
+  // - contract.dateAdded(...)
+  // - contract.dateAuthenticated(...)
+  // - contract.didHashToAddress(...)
+  // - contract.isIdentityAdmin(...)
+  // - contract.isOwner(...)
+  // - contract.isPauser(...)
+  // - contract.isRegistered(...)
+  // - contract.isRegistered(...)
+  // - contract.owner(...)
+  // - contract.paused(...)
+  // - contract.removeIdentityAdmin(...)
+  // - contract.whitelistedContracts(...)
+  // - contract.whitelistedCount(...)
+  // - contract.isWhitelisted(...)
+  // - contract.lastAuthenticated(...)
+  // - contract.isDAOContract(...)
+  // - contract.isBlacklisted(...)
+}
+
+export function handleBlacklistRemoved(event: BlacklistRemoved): void {}
+
+export function handleContractAdded(event: ContractAdded): void {}
+
+export function handleContractRemoved(event: ContractRemoved): void {}
+
+export function handleIdentityAdminAdded(event: IdentityAdminAdded): void {}
+
+export function handleIdentityAdminRemoved(event: IdentityAdminRemoved): void {}
+
+export function handleOwnershipTransferred(event: OwnershipTransferred): void {}
+
+export function handlePaused(event: Paused): void {}
+
+export function handlePauserAdded(event: PauserAdded): void {}
+
+export function handlePauserRemoved(event: PauserRemoved): void {}
+
+export function handleUnpaused(event: Unpaused): void {}
+
+export function handleWhitelistedAdded(event: WhitelistedAdded): void {}
+
+export function handleWhitelistedRemoved(event: WhitelistedRemoved): void {}
+
+export function handleAuthenticate(call: AuthenticateCall): void {
+  let entity = ExampleEntity.load(call.transaction.from.toHex())
+  if (entity == null) {
+    entity = new ExampleEntity(call.transaction.from.toHex())
+    entity.count = BigInt.fromI32(0)
+  }
+  entity.count = entity.count + BigInt.fromI32(1)
+  entity.account = call.params.account
+  entity.save()
```

Comment:
what is this entity?

Reply:
I wasn't supposed to submit this one, I put it there for me as an example from the other entity and forgot to remove it, it should have been like this:

export function handleAuthenticate(call: AuthenticateCall): void {

  let citizen = Citizen.load(call.transcation.from.toHex())
  if (citizen == null) {
    citizen = new Citizen(call.transaction.from.toHex())
  }
  citizen.account = call.params.account
  citizen.lastAuthenticated = call.block.timestamp
  citizen.save()
}

Reply:
this was mistakenly submitted, I used this for myself as an example but apparently forgot to delete this part, it should be:

export function handleAuthenticate(call: AuthenticateCall): void {

  let citizen = Citizen.load(call.transcation.from.toHex())
  if (citizen == null) {
    citizen = new Citizen(call.transaction.from.toHex())
  }
  citizen.account = call.params.account
  citizen.lastAuthenticated = call.block.timestamp
  citizen.save()
}